tutorsuperset/templates/superset/apps/pythonpath/openedx_sso_security_manager.py

```diff
@@ -7,39 +7,101 @@
+    Recommended method of passing bearer tokens.
+
+    Authorization: Bearer h480djs93hd8
+    """
+    headers = headers or {}
+    headers['Authorization'] = 'JWT {}'.format(token)
+    return headers
+
+
+def add_to_body(token, body=None):
+    """Add a Bearer Token to the request body.
+
+    access_token=h480djs93hd8
+    """
+    if body is None:
+        body = ''
+    return add_params_to_qs(body, [('access_token', token)])
+
+
+def add_bearer_jwt_token(token, uri, headers, body, placement='header'):
+    if placement in ('uri', 'url', 'query'):
+        uri = add_to_uri(token, uri)
+    elif placement in ('header', 'headers'):
+        headers = add_to_headers(token, headers)
+    elif placement == 'body':
+        body = add_to_body(token, body)
+    return uri, headers, body
+
+
 
 class OpenEdxSsoSecurityManager(SupersetSecurityManager):
 
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.oauth.oauth2_client_cls.client_cls.token_auth_class.SIGN_METHODS.update({
+            "jwt": add_bearer_jwt_token,
+        })
+
     def set_oauth_session(self, provider, oauth_response):
         """
         Store the oauth token in the session for later retrieval.
         """
         res = super().set_oauth_session(provider, oauth_response)
 
         if provider == "openedxsso":
-            session["oauth_token"] = oauth_response
+            session["oauth_token"] = oauth_response["access_token"]
         return res
+    
+    def decoded_user_info(self):
+        return jwt.decode(self.access_token, algorithms=["HS256"], options={"verify_signature": False})
 
     def oauth_user_info(self, provider, response=None):
-        openedx_apis = current_app.config['OPENEDX_API_URLS']
         if provider == 'openedxsso':
-            oauth_remote = self.appbuilder.sm.oauth_remotes[provider]
-            username_url = openedx_apis['get_username']
-            me = oauth_remote.get(username_url).json()
-            username = me['username']
-
-            user_profile_url = openedx_apis['get_profile'].format(username=username)
-            user_profile = oauth_remote.get(user_profile_url).json()
+            user_profile = self.decoded_user_info()
 
-            user_roles = self._get_user_roles(username)
+            user_roles = self._get_user_roles(user_profile.get('preferred_username'))
 
             return {
                 'name': user_profile['name'],
                 'email': user_profile['email'],
-                'id': user_profile['username'],
-                'username': user_profile['username'],
-                'first_name': '',
-                'last_name': '',
+                'id': user_profile['preferred_username'],
+                'username': user_profile['preferred_username'],
+                'first_name': user_profile['family_name'] if user_profile['family_name'] else user_profile['name'],
+                'last_name': user_profile['given_name'],
```

Comment:
Confirmed the issue @mariajgrimaldi noted above.. 
And I think it would be safer to use `user_profile.get` with sensible defaults everywhere of risking `KeyError`s?

```suggestion
                'first_name': user_profile.get('given_name', user_profile.get('name', '')),
                'last_name': user_profile.get('family_name'),
```